The meaningful part of the patch obtained from Kari E. Hurtta
(elm+ME maintainer), which fixes a double-free in parsing of
certain MIME e-mails. The patch should be included in the next
release of elm.

--- melib/mime_decode.c	25 Aug 2007 11:51:51 -0000	2.0
+++ melib/mime_decode.c	28 Oct 2007 07:39:29 -0000	2.2
@@ -2525,11 +2525,13 @@
 	    unlink(fname);           /* Unlink was delayed */
 	    *name = NULL;
 	}
-	free(fname);
-    }
 
-    if (!name)
 	free(fname);
+
+    } else {
+	if (!name)
+	    free(fname);
+    }
     return tmpfp;
 }
 
